x-pack/plugins/security_solution/server/endpoint/routes/resolver/entity.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { RequestHandler } from 'kibana/server';
+import { TypeOf } from '@kbn/config-schema';
+import { validateEntities } from '../../../../common/endpoint/schema/resolver';
+import { ResolverEntityIndex } from '../../../../common/endpoint/types';
+
+/**
+ * This is used to get an 'entity_id' which is an internal-to-Resolver concept, from an `_id`, which
+ * is the artificial ID generated by ES for each document.
+ */
+export function handleEntities(): RequestHandler<unknown, TypeOf<typeof validateEntities.query>> {
+  return async (context, request, response) => {
+    const {
+      query: { _id, indices },
+    } = request;
+
+    /**
+     * A safe type for the response based on the semantics of the query.
+     * We specify _source, asking for `process.entity_id` and we only
+     * accept documents that have it.
+     * Also, we only request 1 document.
+     */
+    interface ExpectedQueryResponse {
+      hits: {
+        hits:
+          | []
+          | [
+              {
+                _source: {
+                  process: {
+                    entity_id: string;
+                  };
+                };
+              }
+            ];
+      };
+    }
+
+    const queryResponse: ExpectedQueryResponse = await context.core.elasticsearch.legacy.client.callAsCurrentUser(
+      'search',
+      {
+        index: indices,
+        body: {
+          // only return process.entity_id
+          _source: 'process.entity_id',
+          // only return 1 match at most
+          size: 1,
+          query: {
+            bool: {
+              filter: [
+                {
+                  match: {
```

Comment:
I think we can switch this to an IDs search: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-ids-query.html

@andrew-goldstein the `_id` that we're passing should be the unique ES document ID (i.e. `_id` field right) right?

![image](https://user-images.githubusercontent.com/56361221/86016279-974e4480-b9f0-11ea-8acd-c5dc27da87eb.png)

Reply:
I'll give this a shot

Reply:
worked just fine.